py/builtinimport.c

```diff
@@ -583,7 +584,8 @@ mp_obj_t mp_builtin___import___default(size_t n_args, const mp_obj_t *args) {
     // Check if there is a weak link to this module
     char umodule_buf[MICROPY_ALLOC_PATH_MAX];
     umodule_buf[0] = 'u';
-    strcpy(umodule_buf + 1, args[0]);
+    strncpy(umodule_buf + 1, qstr_str(module_name_qstr), sizeof(umodule_buf) - 2);
```

Comment:
> There isn't a "str from str or qstr" function

`mp_obj_str_get_str` is that function (although what you have here is basically equivalent).

That said, maybe for consistency we could make this identical to the external-import case.

```c
VSTR_FIXED(umodule_path, MICROPY_ALLOC_PATH_MAX)
vstr_add_char(path, 'u');
vstr_add_str(path, mp_obj_str_get_str(args[0]));
qstr umodule_name_qstr = qstr_from_str(vstr_str(path)); // or qstr_from_strn(path->buf, path->len)
```

Reply:
I've moved both of them to a separate function.